File: app/src/main/java/com/Phase4/pathways/viewmodel/LoggedInViewModel.java
package com.Phase4.pathways.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.google.firebase.auth.FirebaseUser;
import com.Phase4.pathways.model.AppRepository;

public class LoggedInViewModel extends AndroidViewModel {
    private AppRepository authAppRepository;
    private MutableLiveData<FirebaseUser> userLiveData;
    private MutableLiveData<Boolean> loggedOutLiveData;

    public LoggedInViewModel(@NonNull Application application) {
        super(application);

        authAppRepository = new AppRepository(application);
        userLiveData = authAppRepository.getUserLiveData();
        loggedOutLiveData = authAppRepository.getLoggedOutLiveData();
    }

    public void logOut() {
        authAppRepository.logOut();
    }

    public MutableLiveData<FirebaseUser> getUserLiveData() {
        return userLiveData;
    }

    public MutableLiveData<Boolean> getLoggedOutLiveData() {
        return loggedOutLiveData;
    }
}

File: app/src/main/java/com/Phase4/pathways/viewmodel/LoggedInViewModel.kt
package com.Phase4.pathways.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.Phase4.pathways.models.AppRepository
import androidx.lifecycle.MutableLiveData
import com.google.firebase.auth.FirebaseUser

class LoggedInViewModel(application: Application) : AndroidViewModel(application) {
    private val authAppRepository: AppRepository
    val userLiveData: MutableLiveData<FirebaseUser>
    val loggedOutLiveData: MutableLiveData<Boolean>
    fun logOut() {
        authAppRepository.logOut()
    }

    init {
        authAppRepository = AppRepository(application)
        userLiveData = authAppRepository.userLiveData
        loggedOutLiveData = authAppRepository.loggedOutLiveData
    }
}